### PR TITLE
[v5] Replace usage of http/https for requests with fetch api

### PR DESCRIPTION
**Pull request by innerdvations**
### What does it do?

Replace all http/https requests with fetch api requests

- currently only one place in data-transfer but if I find more I'll add them here

### Why is it needed?

We shouldn't be using http/https directly, and it in fact led to [a bug](https://github.com/strapi/strapi/issues/17481) in the past

Also, trying to standardize everything to use fetch api

### How to test it?

Affects export and transfer --from, so both should be tested to ensure they still work

It should work with http and https servers

### Related issue(s)/PR(s)

Prevents issues like this https://github.com/strapi/strapi/issues/17481 from happening again

### CODE REVIEW

File: packages/core/data-transfer/src/strapi/providers/local-source/assets.ts [hunk 1]
```diff
@@ -1,36 +1,40 @@
...
+        },
+        close() {
+          readableStream.end();
+        },
+        abort() {
+          readableStream.destroy();
+        },
+      });
+
+      res?.body?.pipeTo(writable);
```
Comment (Convly) [inline packages/core/data-transfer/src/strapi/providers/local-source/assets.ts:35]: Can't you just pipeTo the passthrough? I believe it extends both writable & readable streams

Comment (Convly) [reply packages/core/data-transfer/src/strapi/providers/local-source/assets.ts:35]: image

Comment (Convly) [reply packages/core/data-transfer/src/strapi/providers/local-source/assets.ts:35]: Summary of offline discoveries regarding :arrow_up: 
- fetch is returning a libdom.ReadableStream (`source`) which is not compatible with stream.Writable(`dest`)
- Using stream.Readable.fromWeb(`source`).pipe(`dest`) doesn't work as fromWeb expects a stream/web.ReadableStream
- Using stream.Readable.fromWeb(new stream/web.ReadableStream(`source`)).pipe(`dest`) seems to be the best option

Comment (innerdvations) [reply packages/core/data-transfer/src/strapi/providers/local-source/assets.ts]: This should be resolved now, thanks!

File: packages/core/data-transfer/src/strapi/providers/local-source/assets.ts [hunk 2]
```diff
@@ -1,36 +1,47 @@
...
 
-  protocolForPath(filepath)
-    .get(filepath, (res) => {
-      if (res.statusCode !== 200) {
-        readableStream.emit(
-          'error',
-          new Error(`Request failed with status code ${res.statusCode}`)
-        );
+  // fetch the image from remote url and stream it
+  fetch(filepath)
```
Comment (innerdvations) [inline packages/core/data-transfer/src/strapi/providers/local-source/assets.ts:17]:
```suggestion
  strapi.fetch(filepath)
```
This should be going through strapi.fetch, but not sure if we have access here. Will update later.

**Review by Bassel17 — APPROVED** [review]
LGTM!

**Pull request merged**